File: clients/ts-axios-superstruct/v2/echo.ts
import { AxiosInstance, AxiosRequestConfig } from 'axios'
import { strParamsItems, strParamsObject, stringify } from './../params'
import * as t from './../superstruct'
import * as models from './models'

export const client = (axiosInstance: AxiosInstance) => {
    return {
        axiosInstance,

        echoBodyModel: async (parameters: {body: models.Message}): Promise<models.Message> => {
            const headers = strParamsObject({
                "Content-Type": "application/json"
            })
            const bodyJson = t.encode(models.TMessage, parameters.body)
            const response = await axiosInstance.post(`/echo/body_model`, bodyJson, {headers: headers})
            switch (response.status) {
                case 200:
                    return Promise.resolve(t.decode(models.TMessage, response.data))
                default:
                    throw new Error(`Unexpected status code ${ response.status }`)
            }
        },
    }
}

File: services/go/developed/v2/echo.go
package v2

import (
	"errors"
	"test-service/generated/v2/echo"
	"test-service/generated/v2/models"
)

type EchoService struct{}

func (service *EchoService) EchoBodyModel(body *models.Message) (*models.Message, error) {
	return nil, errors.New("implementation has not added yet")
}
